Code/Tools/AssetProcessor/native/utilities/ApplicationManagerBase.cpp

```diff
@@ -1555,8 +1555,39 @@ void ApplicationManagerBase::RegisterBuilderInformation(const AssetBuilderSDK::A
                     retryCount++;
                     result = builderRef->RunJob<AssetBuilder::CreateJobsNetRequest, AssetBuilder::CreateJobsNetResponse>(
                         request, response, s_MaximumCreateJobsTimeSeconds, "create", "", nullptr);
-                } while (result == AssetProcessor::BuilderRunJobOutcome::LostConnection &&
-                         retryCount <= AssetProcessor::RetriesForJobNetworkError);
+
+                    // If a lost connection occured, prepare for a retry using an exponential backoff policy
+                    if ((result == AssetProcessor::BuilderRunJobOutcome::LostConnection || 
+                         result == AssetProcessor::BuilderRunJobOutcome::ProcessTerminated ) && (retryCount <= AssetProcessor::RetriesForJobLostConnection))
+                    {
+                        const int delay = 1 << (retryCount-1);
+
+                        // Check if we need a new builder, and if so, request a new one
+                        if (!builderRef->IsValid())
+                        {
+                            AZStd::string oldBuilderId = builderRef->GetUuid().ToString<AZStd::string>();
+                            builderRef.release();
+                            AssetProcessor::BuilderManagerBus::BroadcastResult(builderRef, &AssetProcessor::BuilderManagerBusTraits::GetBuilder, false);
+
+                            AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Lost connection to builder %s. Retrying with a new builder %s (Attempt %d with %d second delay)",
+                                            oldBuilderId.c_str(),
+                                            builderRef->GetUuid().ToString<AZStd::string>().c_str(),
+                                            retryCount+1,
+                                            delay);
+                        } 
+                        else
+                        {
+                            AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Lost connection to builder %s. Retrying (Attempt %d  with %d second delay)",
+                                            builderRef->GetUuid().ToString<AZStd::string>().c_str(),
+                                            retryCount+1,
+                                            delay);
+                        }
+                        AZStd::this_thread::sleep_for(AZStd::chrono::seconds(delay));
+                    }
+
+                } while ((result == AssetProcessor::BuilderRunJobOutcome::LostConnection || 
+                          result == AssetProcessor::BuilderRunJobOutcome::ProcessTerminated) &&
+                          retryCount <= AssetProcessor::RetriesForJobLostConnection);
```

Comment:
this seems identical to the retry logic below, right? Maybe this could be a separate method reused somehow?

Reply:
I mentioned above that this is PR for a cherry pick, but since enough has changed, i'll clean up the code some more